modules/Sources/Features/NHSendFlow/NHSendFlowStore.swift

```diff
@@ -0,0 +1,268 @@
+                state.path.append(
+                    Path.State.review(
+                        .init(
+                            subtotal: state.amountToSend,
+                            memo: state.memo,
+                            recipient: state.recipient
+                        )
+                    )
+                )
+                return .none
+            case .path(.element(id: _, action: .recipient(.scanQRCodeTapped))):
+                state.path.append(Path.State.scan(.init()))
+                return .none
+            case let .path(.element(id: _, action: .scan(.found(code)))):
+                state.recipient = code
+                state.path = StackState(
+                    state.path.map { state in
+                        if case var .recipient(recipientState) = state {
+                            recipientState.recipient = code
+                            return Path.State.recipient(recipientState)
+                        }
+                    
+                        return state
+                    }
+                )
+                let _ = state.path.popLast()
+                return .none
+            case .path(.element(id: _, action: .review(.sendZcashTapped))):
+                guard !state.isSendingTransaction else { return .none }
+                
+                do {
+                    let storedWallet = try walletStorage.exportWallet()
+                    let seedBytes = try mnemonic.toSeed(storedWallet.seedPhrase.value())
+                    let spendingKey = try derivationTool.deriveSpendingKey(seedBytes, 0, networkType)
+                    
+                    state.isSendingTransaction = true
+                    
+                    let memo: Memo? = state.memo.data.isEmpty ? nil : try Memo(string: state.memo.data)
+                    
+                    let recipient = try Recipient(state.recipient.data, network: networkType)
+                    return .run { [state] send in
+                        do {
+                            await send(NHSendFlowReducer.Action.sendTransactionInProgress)
+                            _ = try await sdkSynchronizer.sendTransaction(
+                                spendingKey,
+                                state.amountToSend,
+                                recipient,
+                                memo
+                            )
+                            await send(NHSendFlowReducer.Action.sendTransactionSuccess)
+                        } catch {
+                            await send(NHSendFlowReducer.Action.sendTransactionFailure)
+                        }
+                    }
+                } catch {
+                    return .task { .sendTransactionFailure }
+                }
+            case .continueTapped:
+                var addMemoState = AddMemoReducer.State()
+                addMemoState.memoCharLimit = state.memoCharLimit
```

Comment:
With the current code I believe the `memoCharLimit` might be set to zero.

Reply:
This is set from the parent reducer